Fix return value from get_process_highwater_mark

mcfadden8 opened the pull request:
This function obtains the high watermark from the VmHWM line in
/proc/PID/status.  The value associated with this field is expressed in
kB.  The previous implementation treated this value as the number of
pages causing the return value to be off by a factor of PAGESIZE/1024.

This fix corrects this by treating VmHWM as kB instead.

davidbeckingsale commented on src/umpire/Umpire.cpp:
```diff
@@ -136,9 +136,8 @@ std::size_t get_process_memory_usage_hwm()
-      long page_size{::sysconf(_SC_PAGE_SIZE)};
       ss >> resident_hwm;
-      rval = std::size_t{resident_hwm * page_size};
+      rval = std::size_t{resident_hwm * 1024};
```
Can you add a comment here about the magical 1024?

mcfadden8 replied:
@davidbeckingsale, this is done (I need your review again)

mcfadden8 commented:
LGTM

mcfadden8 committed:
Fix return value from get_process_highwater_mark

This function obtains the high watermark from the VmHWM line in
/proc/PID/status.  The value associated with this field is expressed in
kB.  The previous implementation treated this value as the number of
pages causing the return value to be off by a factor of PAGESIZE/1024.

This fix corrects this by treating VmHWM as kB instead.
